### PR TITLE
fix(storage): add disk cache buffer threshold to mitigate OOM

### PR DESCRIPTION
**Pull request by MrCroxx**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

On the foyer side:
1. The disk cache write model is refined to allow background disk I/O finish interrupt buffer accumulation.
2. Add disk cache write buffer threshold to prevent OOM.

Related foyer PRs:

https://github.com/MrCroxx/foyer/pull/597
https://github.com/MrCroxx/foyer/pull/598

#17591 

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by Li0k — APPROVED** [review]
LGTM

**Review by hzxa21 — APPROVED** [review]

**Comment by MrCroxx** [comment]
There is a new [Common Committee](https://github.com/orgs/risingwavelabs/teams/common). 🤣

File: src/common/src/config.rs [hunk 1]
```diff
@@ -2234,6 +2251,17 @@ pub fn extract_storage_memory_config(s: &RwConfig) -> StorageMemoryConfig {
...
 
+    let block_cache_flush_buffer_threshold_mb = s
+        .storage
+        .data_file_cache
+        .flush_buffer_threshold_mb
+        .unwrap_or(default::storage::block_file_cache_flush_buffer_threshold_mb());
+    let meta_cache_flush_buffer_threshold_mb = s
+        .storage
+        .meta_file_cache
+        .flush_buffer_threshold_mb
```
Comment (stdrc) [inline src/common/src/config.rs:2262]: If you add `file_` to the variables here, you can easily avoid ugly things like `meta_file_cache_flush_buffer_threshold_mb: meta_cache_flush_buffer_threshold_mb ` below🤡

**Review by stdrc — APPROVED** [review]
Rest lgtm in `common` crate.

File: src/common/src/config.rs (hunk 1 above)
Comment (MrCroxx) [reply src/common/src/config.rs:2262]: Oh I forgot this one. I've already fixed some. 🙏

**Pull request merged**